### ISSUE TITLE
Feature: Install dependencies listed in the manifest

### ISSUE THREAD

**Issue by vvambo** [Trojan295/eso-addons#43]
The dependencies are already parsed, so it should not be too hard to parse them just after the install process, to start installing the needed dependencies.

Optional dependencies could be added with a  '-o' parameter, or maybe optional dependencies are installed by default so --no-optionaldependencies would avoid them being installed.

The function could also take multiple addon names, to run a "reapair" of dependencies for addons that were installed in another way, or before the feature was added, or to install optional ones after the initial installation.

**Comment by Trojan295** [comment]
The problem I see is that the addon description has the name of the dependency, but not a link. I'm not sure the search functionality on esoui.com is reliable enough to find the required addon based only on the name.

E.g. if an addon needs the `LibAddonMenu` and you search for it, you will get multiple results: https://www.esoui.com/downloads/search.php?search=libaddonmenu.

We could try to develop some heuristic to find the correct addon (e.g. the name MUST be the same), but I don't think this is reliable and will work for all cases.

**Comment by vvambo** [comment]
Yeah, if there's no API access, the search function #42 proposed here would have to be called, and selected via User input.

The program could also "remember" those selection, to compose a list that could be added with later versions.
(So that the program checks the list first, and if the keyword fits exactly (+- upper/lower cases) it will directly take the URL from the list.
That'd be ugly but might work (and involve community :D)

**Comment by mahoromax** [comment]
Addons with dependencies often Link their dependencies in the description page.
This would be scrapping again, but if there's a dependency found in the manifest, and the search doesnt give an exact result, it could first scrap the addon page for a Link with the AddonTitle as text, and then go back to the "scrap search result" or "get user to find the right URL" behaviour.

**Comment by vvambo** [comment]
Addons with dependencies often Link their dependencies in the description page.
This would be scrapping again, but if there's a dependency found in the manifest, and the search doesnt give an exact result, it could first scrap the addon page for a Link with the AddonTitle as text, and then go back to the "scrap search result" or "get user to find the right URL" behaviour.

**Comment by buldezir** [comment]
lets trade, you will help me with learning rust (i am php guy :D ), and i will provide you with nice json addons API
